mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1077,58 +1201,53 @@ private class LogisticCostFun(
     fitIntercept: Boolean,
     standardization: Boolean,
     featuresStd: Array[Double],
-    featuresMean: Array[Double],
-    regParamL2: Double) extends DiffFunction[BDV[Double]] {
+    regParamL2: Double,
+    multinomial: Boolean,
+    standardize: Boolean) extends DiffFunction[BDV[Double]] {
 
   override def calculate(coefficients: BDV[Double]): (Double, BDV[Double]) = {
-    val numFeatures = featuresStd.length
     val coeffs = Vectors.fromBreeze(coefficients)
-    val n = coeffs.size
+    val bcCoeffs = instances.context.broadcast(coeffs)
     val localFeaturesStd = featuresStd
-
+    val numFeatures = localFeaturesStd.length
+    val numFeaturesPlusIntercept = if (fitIntercept) numFeatures + 1 else numFeatures
 
     val logisticAggregator = {
       val seqOp = (c: LogisticAggregator, instance: Instance) =>
-        c.add(instance, coeffs, localFeaturesStd)
+        c.add(instance, bcCoeffs.value, localFeaturesStd)
       val combOp = (c1: LogisticAggregator, c2: LogisticAggregator) => c1.merge(c2)
 
       instances.treeAggregate(
-        new LogisticAggregator(numFeatures, numClasses, fitIntercept)
+        new LogisticAggregator(numFeatures, numClasses, fitIntercept, multinomial, standardize)
```

Comment:
In testing I found that the extra divisions required to standardize the data one every iteration have significant impacts for large numbers of features/classes. I changed the `LogisticAggregator` to _optionally_ standardize the data. This way we can change binary LR if needed and/or we can give the option to users so that if they already have standardized data then the runtime will be less.